### ISSUE TITLE
Hotspot image connector new functionality issues/suggestions

### ISSUE THREAD

**Issue by FayCross** [thexerteproject/xerteonlinetoolkits#1303]
- It's confusing that every hotspot automatically shows the editor field that allows you to customise the hotspot icon, even if the hotspot is a polygon & not an icon. Can this only show in the editor when the icon mode has been used?
- The hotspot icons aren't accessible - I can't tab to them and select via keyboard only
- Should the 'new tab' optional property be an extension of the external URL instead of its own property? It's a bit confusing as we now have new window, new tab & external URL. It would make more sense to have a drop down menu called 'open in' as part of external URL that had the option to choose between this tab and new tab.
- I think just 'bubble' would work better that 'nearby bubble' as description of the new option in the display in drop down menu
- I'm not sure the 'on hover' option really works for the lightbox and dialog display options. With the lightbox it will open on hover but then when I close the lightbox it always immediately reopens again even if I am no longer hovering over the hotspot. Perhaps have the display method option show only when bubble is selected

**Comment by ronm123** [comment]
@FayCross @torinfo @ingedonke
I’ve tested this now and I actually like the difference between a 'nearby bubble' vs a centralised lightbox or dialogue. Like Fay I'm not sure that 'nearby' or 'bubble' are the best terms to use. I think mostly this will be used for display of more than tooltip text so I don’t think ‘tooltip’ is appropriate either. I have projects where the hotspot opens a text option with quite a bit of text and relevant images and in testing that sort of thing with the three different 'display in' options I also notice the bubble options forces the text to be aligned centre which I don’t think it should do especially as the other two options don’t and leave that to the formatting of the text.
Example where the 'nearby bubble' is forcing text to be aligned centre regardless of how the text is formatted by the author, which it shouldn't do.
![image](https://github.com/user-attachments/assets/43c7a986-47b1-4c26-9e8e-7f9239465a44)

I wonder if 'relative pop-up' would be a better term than nearby bubble?
HTH
Ron

**Comment by FayCross** [comment]
@torinfo This works a lot better and I do really like the bubbles for showing some content. There are a few issues though:

- [ ] audio does not work when set to open in a bubble - nothing happens on click. If this is intentional then the option shouldn't be there when audio is selected
- [ ] on hover does not work with audio - you always need to click hotspot to open the content. I guess this is intentional as you wouldn't be able to access the audio controls (unless set to autoplay) but the option shouldn't be in the editor if it won't work
- [ ] video - no option to set where it opens so it's always shown in a dialog box
- [ ] when on hover is used and hotspots are tabbed to, you still need to press space bar to open the content. I think when hover is used the content should open on focus too
- [ ] As mentioned in original comments above, I don't think on hover should work when dialog or lightbox are selected. On hover works well with bubbles as they disappear when you move focus away from the hotspot but dialogs and lightboxes need to be manually closed so it's a bit annoying when they open on hover when you don't intend for them to.

**Comment by FayCross** [comment]
New issue with hotspot image connector page

- [ ] Bubbles are shown beneath the side bar so you can't see all the content

**Comment by FayCross** [comment]
- [ ] Bubbles do not always show in an appropriate colour for the theme in use (this is particularly problematic when using the themes available in accessibility options)
